compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -1205,7 +1225,12 @@ class TreeUnpickler(reader: TastyReader,
 
       def completeSelect(name: Name, sig: Signature, target: Name): Select =
         val qual = readTree()
-        val denot = accessibleDenot(qual.tpe.widenIfUnstable, name, sig, target)
+        val denot0 = accessibleDenot(qual.tpe.widenIfUnstable, name, sig, target)
+        val denot =
+          if unpicklingJava && name == tpnme.Object && denot0.symbol == defn.ObjectClass then
+            defn.FromJavaObjectType.denot
```

Comment:
What about imported `Ident`? Also, this needs to be cancelled if it appears in an `extends` clause.

Reply:
If I explicitly put `extends Object` in a Java source, then it will actually be `extends <FromJavaObject>` after typer

Reply:
>What about imported Ident?
java tasty doesnt have imports?

Reply:
If a method takes as argument `java.lang.Object` or `Object`, I expect them to behave the same. Will it?

Reply:
in local testing this will be typed explicitly in TASTy as  `<FromJavaObject>`